### DISCUSSION TITLE
Copy to postgres binary dump

### DISCUSSION THREAD

**Discussion by adriangb** [duckdb/duckdb#11438] [Q&A]
Is there any way to use DuckDB to copy to a postgres binary dump file? The equivalent of copying to a table and then doing `COPY ... WITH FORMAT BINARY` from the table. Since DuckDB already converts it's internal table to this format to copy to the table I'd think it should be possible.

One option for an API would be something like COPY mytable TO STDOUT (FORMAT POSTGRES) and thus COPY 'file.parquet' TO STDOUT (FORMAT POSTGRES).

As an example use case I made a library for this that DuckDB could then replace and it has 200+ stars: https://github.com/adriangb/pgpq

**Comment by Tishj** [comment]
> Since DuckDB already converts it's internal table to this format

I'm not sure I follow? Can you post any sources explaining what it is exactly you mean, supporting this claim?

**Reply by adriangb** [reply]
https://duckdb.org/2022/09/30/postgres-scanner.html#implementation

>  we use the rarely-used binary transfer mode of the Postgres client-server protocol

**Reply by adriangb** [reply]
Basically DuckDB already knows how to do the conversion. What I'm asking for is that there be a way to do pipe the data to a file / stdout instead of over a socket to a postgres database.

**Reply by adriangb** [reply]
@Tishj does that address your question?

**Reply by Tishj** [reply]
I wonder if the statements in that article still hold true, as there have been quite some changes on the repo
<https://github.com/duckdb/postgres_scanner/tree/main>

My expectancy is that this is outdated, as work has been done to support connecting to postgres databases over the network, which I believe would not be possible if we would still use the binary transfer mode?

**Reply by Tishj** [reply]
Either way, this issue is more suited for the postgres_scanner repo than the main DuckDB repo

**Reply by adriangb** [reply]
> work has been done to support connecting to postgres databases over the network, which I believe would not be possible if we would still use the binary transfer mode?

that's definitely not correct. most postgres drivers support binary transfers and they all operate across the network. binary transfer mode is just an encoding scheme for data sent over the wire.

> Either way, this issue is more suited for the postgres_scanner repo than the main DuckDB repo

I'm happy to move the discussion there

**Reply by adriangb** [reply]
@Tishj are you able to transfer the discussion? GitHub doesn't let me, I'd have to open a new one

**Reply by adriangb** [reply]
I opened https://github.com/duckdb/postgres_scanner/issues/204

**Reply by adriangb** [reply]
This has now been implemented: https://github.com/duckdb/postgres_scanner/pull/205
